File: Java/Transacciones/src/practica01/Empresa.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package practica01;

/**
 *
 * @author proyecto
 */
public class Empresa {
    private String razonSocial;
    private String ruc;
    private int pasos;

    public Empresa(String razonSocial, String ruc) {
        this.razonSocial = razonSocial;
        this.ruc = ruc;
    }
    
    public void registrarPasos(int pasos) {
        this.pasos = pasos;
    }

    public int calcularPago(){
        int pago = 0;
        if (this.pasos <= 1000) {
            return 500;
        }
        int excedido = this.pasos - 1000;
        pago = (int) (500 + ((Math.ceil((excedido-1) / 100)+1)*20));
        return pago;
    }
}

File: Ruby/Transacciones/Empresa.rb
class Empresa
    def initialize(razon_social, ruc)
        @razon_social = razon_social
        @ruc = ruc
        @pasos = 0
    end

    def registrar_pasos(pasos)
        @pasos = pasos
    end

    def calcular_pago
        if @pasos <= 1000
            return 500
        end
        pasos_adicionales = ((@pasos - 1000) / 100.00).ceil
        return 500 + pasos_adicionales * 20
    end
end
